### PR TITLE
plan: new plan support hash aggregation.

### PR DESCRIPTION
**Pull request by hanfei1991**
Now we only support hash aggregation.
@shenli @coocood @zimulala @tiancaiamao PTAL

### CODE REVIEW

File: plan/new_physical_plan_builder.go [hunk 1]
```diff
@@ -456,3 +456,35 @@ func splitConditionsByIndexColumns(conditions []expression.Expression, schema *e
...
+
+func (p *LogicalAggregation) convert2NewPhysicalPlan(prop *requiredProp) (taskProfile, error) {
+	task, err := p.getTaskProfile(prop)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if task != nil {
+		return task, nil
+	}
+	task, err = p.convert2HashAggregation(prop)
```
Comment (shenli) [inline plan/new_physical_plan_builder.go]: We will support streaming aggregation latter?

File: plan/task_profile.go [hunk 2]
```diff
@@ -276,3 +282,89 @@ func (sel *Selection) attach2TaskProfile(profiles ...taskProfile) taskProfile {
...
+			return
+		}
+	}
+	_, _, remained := ExpressionsToPB(sc, p.GroupByItems, client)
+	if len(remained) > 0 {
+		return
+	}
+	partialAgg = p.Copy().(*PhysicalAggregation)
+	// TODO: It's toooooo ugly here. Refactor in the future !!
+	gkType := types.NewFieldType(mysql.TypeBlob)
```
Comment (shenli) [inline plan/task_profile.go]: This is ported from old code?

File: plan/new_physical_plan_builder.go (hunk 1 above)
Comment (hanfei1991) [inline plan/new_physical_plan_builder.go]: yeah

File: plan/task_profile.go (hunk 2 above)
Comment (hanfei1991) [inline plan/task_profile.go]: yeah

**Comment by shenli** [comment]
LGTM
@zimulala @winoros @coocood PTAL

File: plan/task_profile.go [hunk 3]
```diff
@@ -296,3 +302,89 @@ func (sel *Selection) attach2TaskProfile(profiles ...taskProfile) taskProfile {
...
+	gkType.Charset = charset.CharsetBin
+	gkType.Collate = charset.CollationBin
+	partialSchema := expression.NewSchema(&expression.Column{RetType: gkType, FromID: p.id, Index: 0})
+	cursor := 0
+	newAggFuncs := make([]expression.AggregationFunction, len(finalAgg.AggFuncs))
```
Comment (coocood) [inline plan/task_profile.go]: `finalAggFunc` is a more consistent name.

**Comment by coocood** [comment]
rest LGTM

**Pull request merged**